File: Leetcode/1750. Minimum Length of String After Deleting Similar Ends/solution.cpp
class Solution {
public:
    int minimumLength(string s) {
        int l = 0, r = s.length() - 1;
        char cur = '\0';
        while (r > l) {
            if (s[l] == cur) {
                l++;
            } else if (s[r] == cur) {
                r--;
            } else if (s[l] == s[r]) {
                cur = s[l];
            } else {
                break;
            }
        }
        return (r == l && r > 0 && s[l] == s[l - 1]) ? 0 : r - l + 1;
    }
};

File: Leetcode/1750. Minimum Length of String After Deleting Similar Ends/solution.rs
impl Solution {
    pub fn minimum_length(s: String) -> i32 {
        let mut l = 0;
        let mut r = s.len() - 1;
        let mut cur: char = '\0';
        while r > l {
            if s.chars().nth(l).unwrap() == cur {
                l += 1;
            } else if s.chars().nth(r).unwrap() == cur {
                r -= 1;
            } else if s.chars().nth(l).unwrap() == s.chars().nth(r).unwrap() {
                cur = s.chars().nth(l).unwrap();
            } else {
                break;
            }
        }
        if r == l && r > 0 && s.chars().nth(l).unwrap() == s.chars().nth(l - 1).unwrap() {
            return 0;
        } else {
            return (r - l + 1) as i32;
        }
    }
}
